### DISCUSSION TITLE
How to use ssh reverse port forwarding?

### DISCUSSION THREAD

**Discussion by anirudhmalik** [anirudhmalik/xhunter#18] [General]
How to use `ssh reverse port forwarding` 

- setup your ubuntu server (vps or aws ec2 with security rules set to `allow all` inbound and outbound connect.
- after success login to your ubuntu server using`ssh -i your_key.pem ubuntu@<remote_server_ip>`
- then after login setup `sudo passwd ubuntu` 
- in /etc/ssh/sshd_config edit `GatewayPorts no` to `GatewayPorts yes` and also edit `PasswordAuthentication no` to `PasswordAuthentication yes`
- then `sudo service sshd restart`
- now your use your `<remote_server_ip>` as you IP
- `ubuntu or ec2user` as username
-  and your `*************` as password to tunnel your device to server.
- 
![IMG_20220718_191428](https://user-images.githubusercontent.com/70255938/179525125-adbf4c8c-a2e2-4ec6-b704-1d23c5cc5d84.jpg)

**Comment by Ruben978** [comment]
What if I don't have aws, azure or any vps

Is there any free way?

**Reply by anirudhmalik** [reply]
[localtunnel](https://github.com/localtunnel/localtunnel), but recently it is facing some issue. their server seems to be down

**Reply by hammadhaxxcker-stack** [reply]
Bruh this payload work after close payload app or not work listning

**Comment by GLITCHMaker84645** [comment]
لماذا الهيركو أصبح مدفوعا

**Comment by GLITCHMaker84645** [comment]
نريد طريقه مجانيه للحقن

**Comment by Assam100** [comment]
نريد طريقه مجانية للحقن

**Comment by ALAMIN696969** [comment]
Your explanation is so terrible bro can you make a video to explain this  we have no option to hack Android your app because your heroku server need to pay next update you change heroku please and your custom server how to create we have a no idea please make a video 🥲

**Comment by Sahasaalim** [comment]
Bro can you pls make a video teaching the custom method. The heroku requires payment and your explanation here too doesn't add up

**Comment by anirudhmalik** [comment]
Closing in favor of v2. See [v2.0-demo](https://github.com/anirudhmalik/xhunter/releases/tag/v2.0-demo) and the repo README. New bugs: open a fresh issue with repro steps. Thanks.

**Reply by ALAMIN696969** [reply]
Bro I download it your new version it is good but how to use properly make a video and and tell us about the all the features your try to in the future and making proper video and explain everything how to do and what type of feature will future we get okay make it English 😁 thank you for the new version and working on this app 😘
